stock_move_ignore_last_purchase_date/models/stock_move.py

```diff
@@ -0,0 +1,12 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    ignore_last_purchase_date = fields.Boolean(
+        related="picking_id.ignore_last_purchase_date", store=True
```

Comment:
Updating this field doesn't automatically trigger the update of `last_purchase_date` of the product.  We may make this a compute field, and call `_compute_last_purchase_date()` inside the compute method.